backend/audit/test_schemas.py

```diff
@@ -403,76 +391,81 @@ def test_all_booleans(self):
+                ls = random.sample(gaap_results, n)
                 simple_case["gaap_results"] = ls
-                validate(simple_case, self.AUDIT_INFO_SCHEMA)
-
-    def test_all_framework_basis(self):
-        simple_case = jsoncopy(self.SIMPLE_CASE[1])
+                validate(simple_case, schema)
 
-        sp_framework_basis = [
-            "cash_basis",
-            "tax_basis",
-            "contractual_basis",
-            "other_basis",
-        ]
-
-        for _ in range(5):
-            for n in range(1, 5):
-                opinions = random.sample(self.ALL_OPINIONS, n)
-                basis = random.sample(sp_framework_basis, n)
-                simple_case["sp_framework_opinions"] = opinions
-                simple_case["sp_framework_basis"] = basis
-                validate(simple_case, self.AUDIT_INFO_SCHEMA)
+        # Test when it is 'not_gaap', which requires other fields.
+        simple_case["gaap_results"] = ["not_gaap"]
+        simple_case["is_sp_framework_required"] = random.choice([True, False])
+        simple_case["sp_framework_basis"] = random.choices(
+            ["cash_basis", "tax_basis", "contractual_basis", "other_basis"], k=2
+        )
+        simple_case["sp_framework_opinions"] = random.choices(
+            [
+                "unmodified_opinion",
+                "qualified_opinion",
+                "adverse_opinion",
+                "disclaimer_of_opinion",
+            ],
+            k=2,
+        )
+        validate(simple_case, schema)
 
     def test_bad_gaap_results(self):
+        schema = self.AUDIT_INFO_SCHEMA
         simple_case = jsoncopy(self.SIMPLE_CASE[0])
+        not_gaap_values = [
+            "state",
+            "local",
+            "tribal",
+            "higher-ed",
+            "non-profit",
+            "unknown",
+            "none",
+        ]
 
-        for _ in range(5):
-            for n in range(1, 5):
-                ls = random.sample(self.BAD_VALUES, n)
-                simple_case["gaap_results"] = ls
-                self.assertRaises(
-                    exceptions.ValidationError,
-                    validate,
-                    simple_case,
-                    self.AUDIT_INFO_SCHEMA,
-                )
-
-    def test_bad_framework_basis(self):
-        simple_case = jsoncopy(self.SIMPLE_CASE[1])
-
-        for _ in range(5):
-            for n in range(1, 5):
-                bad_values = random.sample(self.BAD_VALUES, n)
-                simple_case["sp_framework_opinions"] = bad_values
-                simple_case["sp_framework_basis"] = bad_values
-                self.assertRaises(
-                    exceptions.ValidationError,
-                    validate,
-                    simple_case,
-                    self.AUDIT_INFO_SCHEMA,
-                )
+        for word in not_gaap_values:
+            simple_case["gaap_results"] = [word]
+            self.assertRaises(exceptions.ValidationError, validate, simple_case, schema)
```

Comment:
Any reason why we'll want to test bad values for `gaap_results` only but not for  `sp_framework_opinions` and `sp_framework_basis` ? @danswick @tadhg-ohiggins

Reply:
Good point. I added two very simple tests below.